### ISSUE TITLE
numpy version bound is incorrect

### ISSUE THREAD

**Issue by anishathalye** [cleanlab/cleanlab#698]
Currently, setup.py [says](https://github.com/cleanlab/cleanlab/blob/f77ef9f0651292d79b880addac4eff60673d955f/setup.py#L101) that this package requires `numpy>=1.20.0`.

Using `numpy==1.20.3`, if I `import cleanlab`, it fails with:

```
AttributeError: module 'numpy.typing' has no attribute 'NDArray'
```

[`numpy.typing.NDArray`](https://numpy.org/devdocs/reference/typing.html#numpy.typing.NDArray) was introduced in 1.21, so the version bound that this package uses should be at least that.

Just FYI, the [latest Databricks runtime](https://docs.databricks.com/release-notes/runtime/11.3.html) ships with `numpy==1.20.3` right now, this means that users on Databricks can `!pip install cleanlab` (which appears to work fine) and then `import cleanlab` and it will fail.

**Comment by huiwengoh** [comment]
I believe @elisno just addressed this in #688, could you try installing from master branch?

**Comment by anishathalye** [comment]
That works, thank you!

(I should have searched the issues/PRs first...)
